### ISSUE TITLE
Include Admonition "example" qualifier

### ISSUE THREAD

**Issue by harveyslash** [squidfunk/mkdocs-material#646]
## Description

The docs say admonition allows for custom qualifier, and mkdocs-material adds a few of its own. 
There is no example to use a custom icon and color for the blocks. 

I would like to use 5 new icons for documenting a RESTful api(and access them with my own qualifier), but couldn't find any documentation on how to do so

**Comment by facelessuser** [comment]
I just took a look at the Material style to figure out how to add my own: https://github.com/facelessuser/pymdown-extensions/blob/master/docs/src/scss/_admonition.scss.  Also, details should inherit the same stuff: https://github.com/facelessuser/pymdown-extensions/blob/master/docs/src/scss/_details.scss.

I'm no web developer, so if you see a better approach, feel free to chime in, but it seems to work for my purposes:  https://facelessuser.github.io/pymdown-extensions/extensions/details/

**Comment by harveyslash** [comment]
thanks for replying so quickly. 
I am not familiar with this, so I am not sure how to add your file. 
Right now I have just a docs folder, and the yml config file.

How can I configure it like you have?

**Comment by facelessuser** [comment]
Probably you avoid use SCSS and just use CSS to avoid having to transpile it over to CSS.  I just ran my theme through an unminifier: https://github.com/facelessuser/pymdown-extensions/blob/master/docs/theme/extra-4e53a56959.css.

You would be particularly interested in the following.  Just an explanation, I am adding something called `settings` and `example`.

```css
.md-typeset .admonition.settings, .md-typeset details.settings {
    border-left: .4rem solid #a0f;
}
.md-typeset .admonition.settings>.admonition-title, .md-typeset details.settings>.admonition-title, .md-typeset details.settings>summary {
    border-bottom: .1rem solid rgba(170, 0, 255, .1);
    background-color: rgba(170, 0, 255, .1);
}
.md-typeset .admonition.settings>.admonition-title:before, .md-typeset details.settings>.admonition-title:before, .md-typeset details.settings>summary:before {
    color: #a0f;
    content: "settings"}
.md-typeset .admonition.example, .md-typeset details.example {
    border-left: .4rem solid #6200ea;
}
.md-typeset .admonition.example>.admonition-title, .md-typeset details.example>.admonition-title, .md-typeset details.example>summary {
    border-bottom: .1rem solid rgba(98, 0, 234, .1);
    background-color: rgba(98, 0, 234, .1);
}
.md-typeset .admonition.example>.admonition-title:before, .md-typeset details.example>.admonition-title:before, .md-typeset details.example>summary:before {
    color: #6200ea;
    content: "format_list_numbered"}
```

You can use that as an example and put it in your own CSS file and use the `extra_css` in your `mkdocs.yml` file to add the changes like I do: https://github.com/facelessuser/pymdown-extensions/blob/master/mkdocs.yml#L116.

**Comment by harveyslash** [comment]
Thanks! when I added that, im able to see the gear icon. How can I add my custom icons ?

**Comment by facelessuser** [comment]
Material theme uses fontawesome for the icons in the admonitions and details objects.  So you can notice that I have a couple of selectors that target specific element and set the `:before` to `content: "settings"` or `content: "format_list_numbered"` for settings and examples respectively.  These are aliases that get resolved to the appropriate Unicode code point for font awesome.  You can look up the names over at: http://fontawesome.io/icons/.  I guess in my case, I'm using `settings`, but I could have used `cog` or `gear` as they are aliases.  Play around with the style by plugging in other fontawesome character names.

**Comment by harveyslash** [comment]
any way to use something like my own images as fonts? 
font awesome may not have all the icons that I want to put

**Comment by facelessuser** [comment]
Well, not images as a font, but I'm sure you can use an image instead of a font for your specific cases.  You'll just have to play with the CSS to come up with your own solution.

**Comment by harveyslash** [comment]
I tried changing 'settings' to 'download', but it just shows a blank space now. 

my _admonition.scss looks like : 
      download config: $clr-purple-a700 "download",
and my css looks like :


.md-typeset .admonition.download, .md-typeset details.download {
    border-left: .4rem solid #a0f;
}
.md-typeset .admonition.download>.admonition-title, .md-typeset details.download>.admonition-title, .md-typeset details.download>summary {
    border-bottom: .1rem solid rgba(170, 0, 255, .1);
    background-color: rgba(170, 0, 255, .1);
}
.md-typeset .admonition.download>.admonition-title:before, .md-typeset details.download>.admonition-title:before, .md-typeset details.download>summary:before {
    color: #a0f;
    content: "download"}

**Comment by harveyslash** [comment]
Never mind, the correct name seems to be arrow_downward , which I found from here : 
https://docs.flutter.io/flutter/material/Icons/arrow_downward-constant.html

**Comment by makshh** [comment]
Yes, Material for MkDocs uses Material Icons. FontAwesome icons are used only for social icons.

You can find complete list of Material Icons names here:

https://material.io/icons/

**Comment by facelessuser** [comment]
Ha, totally forgot those were Material, oh well :).

**Comment by squidfunk** [comment]
I'm always happy to include more of the admonition styles if they make any sense. I also think about including your example styles, @facelessuser. Feel free to propose something for rest and we see if it makes any sense to integrate :-)

**Comment by facelessuser** [comment]
Adding `example` would be great, it's probably the one I'd like the most to be added as it is something I'd use a lot, and the usage of it is distinct enough that I think warrants a style devoted to it.  I kind of settled on the numerical list icon as it kind of inferred steps, but I'm open to if someone has something better.  It was difficult trying to convey "examples", but steps seemed to make sense, at least to me.

As for any other styles, I play around with ideas here and there, but nothing that I'm committed to like I am with `example`.

**Comment by squidfunk** [comment]
The numerical list icon is the reason I love the example qualifier for, as for me it's really self explanatory.

Also, I would love to have some way to document a REST API, so little icons for GET, PUT, DELETE etc. We could do this as an extension to admonition/details. Is there some standard Markdown target for API documentation by a generator like swagger? We could try and optimize Material a little so it nicely renders the output Markdown by such a generator.

**Comment by harveyslash** [comment]
Documenting a rest api is the very reason I wanted to add custom icons. 
so far i am using the following icons for it : 


```
.md-typeset .admonition.get_req, .md-typeset get_req.download {
    border-left: .4rem solid #A5D6A7;
}

.md-typeset .admonition.get_req > .admonition-title, .md-typeset details.get_req > .admonition-title, .md-typeset details.get_req > summary {
    border-bottom: .1rem solid #388E3C;
    background-color: rgba(0, 170, 0, .1);
}

.md-typeset .admonition.get_req > .admonition-title:before, .md-typeset details.get_req > .admonition-title:before, .md-typeset details.get_req > summary:before {
    color: #558B2F;
    content: "arrow_downward"
}

.md-typeset .admonition.post_req, .md-typeset post_req.download {
    border-left: .4rem solid #E1BEE7;
}

.md-typeset .admonition.post_req > .admonition-title, .md-typeset details.post_req > .admonition-title, .md-typeset details.post_req > summary {
    border-bottom: .1rem solid #E1BEE7;
    background-color: #E1BEE7;
}

.md-typeset .admonition.post_req > .admonition-title:before, .md-typeset details.post_req > .admonition-title:before, .md-typeset details.post_req > summary:before {
    color: #4A148C;
    content: "arrow_upward"
}

.md-typeset .admonition.custom_req, .md-typeset custom_req.download {
    border-left: .4rem solid #6D4C41;
}

.md-typeset .admonition.custom_req > .admonition-title, .md-typeset details.custom_req > .admonition-title, .md-typeset details.custom_req > summary {
    border-bottom: .1rem solid #6D4C41;
    background-color: #A1887F;
}

.md-typeset .admonition.custom_req > .admonition-title:before, .md-typeset details.custom_req > .admonition-title:before, .md-typeset details.custom_req > summary:before {
    color: #FAFAFA;
    content: "code"
}


.md-typeset .admonition.del_req, .md-typeset del_req.download {
    border-left: .4rem solid #EF5350;
}

.md-typeset .admonition.del_req > .admonition-title, .md-typeset details.del_req > .admonition-title, .md-typeset details.del_req > summary {
    border-bottom: .1rem solid #FFCDD2;
    background-color: #FF5252;
}

.md-typeset .admonition.del_req > .admonition-title:before, .md-typeset details.del_req > .admonition-title:before, .md-typeset details.del_req > summary:before {
    color: #FAFAFA;
    content: "delete_forever"
}

```
Example: 
<img width="722" alt="screen shot 2017-12-21 at 6 04 55 am" src="https://user-images.githubusercontent.com/7107410/34253320-0f82543a-e615-11e7-8355-456eaabff6a9.png">

**Comment by facelessuser** [comment]
> The numerical list icon is the reason I love the example qualifier for, as for me it's really self explanatory.

Nice :).

> Is there some standard Markdown target for API documentation by a generator like swagger? We could try and optimize Material a little so it nicely renders the output Markdown by such a generator.

I don't know what you mean by "target".  Are you referring to like a class name?  Or some specific Markdown syntax? Unfortunately I'm not familiar with what Swagger is.

**Comment by jsbUSMC** [comment]
Browsing through the [Swagger Codegen repo](https://github.com/swagger-api/swagger-codegen/tree/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages) you can see all the languages that they've got a target for, and unfortunately, Markdown isn't among them. However, they do have generators for targeting `swagger.json` and `swagger.yaml`. Looks like their codegen project is Java 8. I don't know if you'd want to have a parser for swagger files, or if you'd like to maybe see how [Slate](https://github.com/lord/slate) does things, which is a Ruby static API documentation project.

**Comment by squidfunk** [comment]
Thanks @jsbUSMC for your input, but I think for now it's out of scope. Maybe at some point later.

**Comment by squidfunk** [comment]
Closed accidentally. Reopening this to include @facelessuser `example` qualifier.

**Comment by squidfunk** [comment]
Added `example` to Admonition styles and released as part of 2.3.0.
